src/main/java/edu/hm/hafner/coverage/CyclomaticComplexity.java

```diff
@@ -32,7 +32,7 @@ public CyclomaticComplexity(final int complexity, final Metric metric) {
     }
 
     @Override
-    protected IntegerValue create(final int value) {
-        return new CyclomaticComplexity(value);
+    protected IntegerValue create(final int value, final Metric metric) {
+        return new CyclomaticComplexity(value, metric);
```

Comment:
Yes, the easiest would be to just rename `CyclomaticComplexity` to `MetricValue` and remove the Constructor without the enum parameter. The `coverage-plugin` would also need to be changed and this would be a breaking change so this would be a major version change.